smallrye-reactive-messaging-jms/src/main/java/io/smallrye/reactive/messaging/jms/tracing/JmsAttributesExtractor.java

```diff
@@ -0,0 +1,114 @@
+package io.smallrye.reactive.messaging.jms.tracing;
+
+import static io.opentelemetry.semconv.SemanticAttributes.*;
+
+import java.util.Collections;
+import java.util.List;
+
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.incubator.semconv.messaging.MessagingAttributesGetter;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+
+public class JmsAttributesExtractor implements AttributesExtractor<JmsTrace, Void> {
+    private final MessagingAttributesGetter<JmsTrace, Void> messagingAttributesGetter;
+
+    public JmsAttributesExtractor() {
+        this.messagingAttributesGetter = new JmsMessagingAttributesGetter();
+    }
+
+    @Override
+    public void onStart(final AttributesBuilder attributes, final Context parentContext, final JmsTrace jmsTrace) {
+        String groupId = jmsTrace.getGroupId();
+        String clientId = jmsTrace.getClientId();
+        if (groupId != null && clientId != null) {
+            String consumerId = groupId;
+            if (!clientId.isEmpty()) {
+                consumerId += " - " + clientId;
+            }
+            attributes.put(MESSAGING_CONSUMER_ID, consumerId);
+        }
+        if (groupId != null) {
```

Comment:
We can remove this one.

Reply:
I am not quite sure i understand this. Do you think we can remove groupId and clientId from the JmsTrace.Builder, so we only have queue and properties? And then cleanup?

Reply:
I meant to suggest removing `attributes.put(MESSAGING_KAFKA_CONSUMER_GROUP, groupId);` because that key is for kafka.

Reply:
I have removed the use of MESSAGING_KAFKA_CONSUMER_GROUP. But i could use your help to get the tests in TestPropagationTest working.